frontend/app/services/subscription-service.server.ts

```diff
@@ -136,6 +108,35 @@ function createSubscriptionService() {
     return cdcpSubscriptions.at(0);
   }
 
+  async function deleteSubscription(userId: string, subscriptionId: string) {
+    const auditService = getAuditService();
+    const instrumentationService = getInstrumentationService();
+    auditService.audit('alert-subscription.delete', { subscriptionId });
+
+    const url = new URL(`${CDCP_API_BASE_URI}/api/v1/users/${userId}/subscriptions/${subscriptionId}`);
```

Comment:
We should be using the HATEOAS links to perform the delete rather than hardcoding the endpoint.

That means that this function should 
1. accept the `userInfoToken.sub` as the `userId` argument
2. call `/users?raoidcUserId=${userId}` and get the response
3. use the `links` in the response to retrieve the URL for the subscription we wish to delete

Reply:
fixed

Reply:
fixed this for real

Reply:
No need to expose the `selfUrl` in the domain model from the service. This field is an implementation detail that only the service should know about. Instead, call GET `/users?raoidcUserId=${userId}` to get the user rather than `getSubscription(..)`.

Reply:
updated